sklearn/preprocessing/tests/test_target_encoder.py

```diff
@@ -712,3 +714,288 @@ def test_pandas_copy_on_write():
     with pd.option_context("mode.copy_on_write", True):
         df = pd.DataFrame({"x": ["a", "b", "b"], "y": [4.0, 5.0, 6.0]})
         TargetEncoder(target_type="continuous").fit(df[["x"]], df["y"])
+
+
+# ---------------------------------------------------------------------------
+# Small batch Target encoder tests
+# ---------------------------------------------------------------------------
+
+
+class _FakeDF:
```

Comment:
we don't use such things, we simply use a dataframe (pandas or polars) and we skip the test if those packages are not installed (check other similar tests please)